Check for `is_numeric()` instead of `is_scalar()` on IntegerType

sdustinh opened the pull request:
Reference: https://github.com/cakephp/cakephp/issues/12509

sdustinh committed:
Check for `is_numeric()` instead of `is_scalar()`.

Fixes issue #12509

lorenzo commented on src/Database/Type/IntegerType.php:
```diff
@@ -98,7 +98,7 @@ public function toPHP($value, Driver $driver)
     {
         foreach ($fields as $field) {
-            if (!isset($values[$field])) {
+            if (!isset($values[$field]) || !is_numeric($values[$field])) {
```
It should throw instead of drop the value

markstory replied:
:+1: we drop values in marshalling, and raise when converting to PHP or DB values.

sdustinh commented:
@lorenzo @markstory If I'm going to throw the exception for non-numeric values in `manyToPHP()`, do you want me to emulate `toDatabase()` and `continue` for `null` and `''` or throw the exception for them as well?

markstory commented:
`null` and `''` should be cast into null and don't require exceptions.

sdustinh committed:
Throw exception on `manyToPHP()` for non-numeric.

markstory commented:
There are a few tests that are failing. Those tests will need to be updated before this can be merged in.

sdustinh commented:
:+1: Just wanted to make sure we were all okay with the changes before fixing the tests.

sdustinh committed:
Fix failing unit tests.

I'm not sure if these changes maintain the original intent of the tests. It looks like they were written specifically to see if the integer would be preserved in a string of text. Either way, the tests have been updated to reflect the framework changes and should pass now.

lorenzo commented on src/Database/Type/IntegerType.php:
```diff
@@ -101,6 +101,14 @@ public function manyToPHP(array $values, array $fields, Driver $driver)
                 continue;
             }
+
+            if ($values[$field] !== null && $values[$field] !== '' && !is_numeric($values[$field])) {
```
* no need for `$values[$field] !== null` there is an `isset` check for the value a few lines above

* no need for `$values[$field] !== ''` the `is_numeric`, we want the `is_numeric` check to catch it

sdustinh replied:
:+1: removed them

sdustinh committed:
Removed some conditionals from review notes.
